api/test/wfl/tools/datasets.clj

```diff
@@ -0,0 +1,45 @@
+(ns wfl.tools.datasets
+  (:require [clojure.string :as str]
+            [clojure.data.json :as json]
+            [wfl.environment :as env]
+            [wfl.service.datarepo :as datarepo]
+            [wfl.service.google.storage :as storage])
+  (:import (java.util UUID)))
+
+(defn unique-dataset-request
+  ""
+  [dataset-basename]
+  (let [profile (env/getenv "WFL_TERRA_DATA_REPO_DEFAULT_PROFILE")]
+    (-> (str "test/resources/datasets/" dataset-basename)
+        slurp
+        json/read-str
+      ;; give it a unique name to avoid collisions with other tests
+        (update "name" #(str % (-> (UUID/randomUUID) (str/replace "-" ""))))
+        (update "defaultProfileId" (constantly profile)))))
+
+(defn ^:private ensure-tdr-can-read [bkt-obj-pairs]
```

Comment:
I think it's better that we don't adjust permissions in code like this. Better we require users to grant permissions than we grant them ourselves (likewise with all IAM-related endpoints)

Add the TDR dev SA to be a reader of this bucket in gotc-deploy
Figure our how TDR production works and do the same so we don't have to test permissions in production.